iceoryx_utils/test/moduletests/test_posix_timer.cpp

```diff
@@ -520,3 +520,24 @@ TIMING_TEST_F(Timer_test, CatchUpPolicySkipToNextBeatCallsLessCallbacksThanASAPT
 
     TIMING_TEST_EXPECT_TRUE(softTimerCounter < asapTimerCounter);
 });
+
+TEST_F(Timer_test, DISABLED_SelfTriggeringTimerWorksAndDoesNotCauseSegFault)
+{
+    Duration selfTriggerTimeout = 1_ns;
```

Comment:
When you start the timer with 1 ns the value is truncated to zero because the cxx::duartion has an truncation error. See https://github.com/eclipse/iceoryx/issues/190

Reply:
No this is not the case here. If you uncomment line 531 the test will work fine and the condition `counter == repetition` will hold. The line 531 causes the seg fault since there seems to be some weird race condition occuring.

Reply:
I assume the timeout is chosen this small but nonzero for a reason (e.g. to trigger almost immediately). I do not know what the variable name is supposed to mean (what is self triggering?). Does the timer not just have a callback in which user code (with some restrictions) can be executed? As far as I see from the test, the timer restarts itself in the callback, but with the ONCE policy. But it is also initially started with this policy.

I have not researched it, but the underlying structure is a POSIX timer. Could it be that the fire ONCE policy causes it to cleanup something in the kernel before the other call has properly finished? Each callback runs in some kind of context/thread, but I assume the ONCE policy is used by the kernel data for the timer to indicate it is not needed after the callback is complete. This might cause the other instance to access an invalid timer struct.
Maybe worth investigating if the cause for the segmentation fault is not yet known...my best guess from my current understanding.

Bottom line: maybe restart is not allowed when RunMode::Once is used.

Reply:
The segfault is documented internally and a issue is already created, see #243 .
The goal of this unit test is to ease the investigation of the segmentation fault when someone starts digging here.
- `restart` is allowed and makes absolute sense when the policy is `RunMode::Once`